Source/Fuse.Nodes/Visual.Children.uno

```diff
@@ -187,33 +226,69 @@ namespace Fuse
 		*/
 		internal void InsertNodes(int index, IEnumerator<Node> items)
 		{
+			InsertNodesImpl(index, items, false);
+		}
+
+		internal void InsertOrMoveNodes(int index, IEnumerator<Node> items)
+		{
+			InsertNodesImpl(index, items, true);
+		}
+		
+		
+		void InsertNodesImpl(int index, IEnumerator<Node> items, bool allowMove)
+		{
 			if (index <0 || index > Children.Count)
 				throw new ArgumentOutOfRangeException("index");
 			
 			//cleanup all nodes first
 			while (items.MoveNext())
 				InsertCleanup( items.Current );
 
+			//becomes non-null on the first moved node
+			HashSet<Node> moved = null;
+			
 			//nodes should be considered added in the same group
 			var capture = CaptureRooting();
 			try
 			{
 				//then add all
 				items.Reset();
 				while (items.MoveNext())
-					_children.Insert(index++, items.Current);
+				{
+					var c = items.Current;
+					if (allowMove)
+					{
+						var where = IndexOf(c);
+						if (where != -1)
+						{
+							_children.RemoveAt(where);
+							if (where < index)
+								index--;
+							if (moved == null)
+								moved = new HashSet<Node>();
+							moved.Add(c);
+						}
+					}
+					_children.Insert(index++, c);
+				}
 
 				//then process them
 				items.Reset();
 				while (items.MoveNext())
-					OnAdded(items.Current);
+				{
+					var c = items.Current;
+					if (moved == null || !moved.Contains(c))
+						OnAdded(c);
+					else
+						OnMoved(c);
+				}
 			}
 			finally
 			{
 				ReleaseRooting(capture);
 			}
 		}
-
+		
```

Comment:
I don't really care so much if it's one way or the other (I have a slight preference for no trailing whitespace, as it just seems pointless), I would prefer if we didn't needlessly change it.